### PR TITLE
Added model zoo 

### PR DESCRIPTION
**Pull request by bveliqi**
From a user's point of view the API is basically this:

```
from foolbox import zoo
model = zoo.get_model(url = 'git@github.com:bethgelab/cifar10_challenge.git')
```

Model owners wanting to make the model foolbox-ready need to put a `foolbox_model.py` containing a simple `create()` function into the root folder  of their git repository.

Besides https://github.com/bethgelab/analysis-by-synthesis-model, I have done this already for three other models:
- https://github.com/bethgelab/convex_adversarial
- https://github.com/bethgelab/mnist_challenge
- https://github.com/bethgelab/cifar10_challenge

Model providers who provide the weights of their models in another place than the git repo itself, get use the following foolbox.zoo utilities to download the weights:

```
from foolbox import zoo

weights_path = zoo.fetch_weights(
        'https://www.dropbox.com/s/ywc0hg8lr5ba8zd/secret.zip?dl=1',
        unzip=True
    )
```

Full example: https://github.com/bethgelab/cifar10_challenge/blob/master/foolbox_model.py#L10

This test here: https://github.com/bveliqi/foolbox/blob/6a54bfafc89723fb9c73293ba7593a93b26a0dfe/foolbox/tests/test_model_zoo.py is more of an integration test than a unit test and depends on the git/weights providers being up & accessible. So I'm not sure whether we want to run these each & every time on our CI systems since they might take a bite more time compared to our other tests until now...

Also, I will have to add the documentation for the model zoo - but you can already review the code while I'll do that.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
     'numpy',
     'scipy',
     'setuptools',
+    'pytest'
```
Comment (bveliqi) [inline setup.py]: delete

File: requirements-dev.txt [hunk 2]
```diff
@@ -9,3 +9,6 @@ flake8 >= 3.3.0
 python-coveralls >= 2.9.1
 pillow >= 4.1.1
 randomgen >= 1.14.4
+requests >= 2.18.4
+GitPython >= 2.1.10
```
Comment (bveliqi) [inline requirements-dev.txt]: add to setup.py not to `-dev.py`

File: foolbox/zoo/weights_fetcher.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        _download(file_path, weights_uri, local_path)
+
+    if unzip:
+        file_path = _extract(local_path, filename)
+
+    return file_path
+
+
+def _filename_from_uri(url):
+    filename = url.rsplit('/', 1)[-1]
```
Comment (bveliqi) [inline foolbox/zoo/weights_fetcher.py]: add comment

Comment (bveliqi) [reply foolbox/zoo/weights_fetcher.py]: use split instead of rsplit

File: requirements-dev.txt [hunk 4]
```diff
@@ -9,3 +9,6 @@ flake8 >= 3.3.0
...
+responses >= 0.9.0
```
Comment (bveliqi) [inline requirements-dev.txt]: add to `-dev.txt`

**Comment by jonasrauber** [comment]
Great job! Once the final changes that we just discussed have been integrated this can be merged. Please don't forget to add comments for the less obvious parts.

**Comment by jonasrauber** [comment]
Looks good! Two comments regarding the example in the docs:
1) You shouldn't use `imagenet_example` on the CIFAR model
2) Are you sure that model takes BGR instead of RGB

**Pull request merged**